public/calendar_reboot.js

```diff
@@ -921,151 +933,168 @@ var LocalCalendarStorage = function(params) {
+                }
+            })
+            .catch(function(err) {
+                self.endWork();
+                console.error(err);
+                return err;
+            })
         })
-        .catch( function() {
-            console.log("getAllCalendarIds catch function running.");
-        });
-
     };
     
     self.save = function(calendarObj) {
         //save an App object (like a calendar object for example) in storage
         
-        //store the state in localStorage
-
-        var stateP = new Promise(function(resolve, reject) {
-            storeInLocalStorage(toKey(calendarObj.state.uniqueId), calendarObj.state);
-            jitter(resolve);
-        });
+        // Store the calendar state in the firebase database.
+        // TO DO save the actual calendar state.
         
+        var userId = self.user.uid;
+        var calUniqueId = calendarObj.state.uniqueId;
+        var calTitle = calendarObj.state.title;
+        var calState = calendarObj.state;
         
-        //put calendar in allCalendarIdss and store it
+        // Store the user's allCalendarIds and their calendarState .. setting the
+        // active calendar has it's own method.
+        var updates = {};
+        updates['users/' + userId + '/allCalendarIds/' + calUniqueId] = calTitle;
+        updates['calendars/' + calUniqueId + '/calendarState/'] = calState;
+        updates['calendars/' + calUniqueId + '/permissionRead/'] = userId;
+        updates['calendars/' + calUniqueId + '/permissionWrite/'] = userId;
         
-        var idsP = self.getAllCalendarIds()
-            .then(function (allCalendarIds) {
-                allCalendarIds[calendarObj.state.uniqueId] = calendarObj.state.title;
-                storeInLocalStorage(toKey(allCalendarIdsKey), allCalendarIds);
-            })
-            .catch(function () {
-                console.log("No previous calendars in storage");
-                var allCalendarIds = {};
-                allCalendarIds[calendarObj.state.uniqueId] = calendarObj.state.title;
-                storeInLocalStorage(toKey(allCalendarIdsKey), allCalendarIds);
-            })
-            
-        return Promise.all([stateP, idsP]);
+        self.startWork();
+        return new Promise(function(resolve, reject) {
+            self.database.ref().update(updates);
+            self.endWork();
+            resolve();  // Still unsure whether I will need to use reject in here or not.
+        })
+    };
+    
+    self.setActiveById = function(calendarObjId) {
+        // Set the active calendar/object by using its Id
+        
+        var userId = self.user.uid;
+        var updates = {};
+        updates['users/' + userId + '/currentActiveCalendar/'] = calendarObjId;
+        
+        self.startWork();
+        return new Promise( function(resolve, reject) {
+            self.database.ref().update(updates);
+            self.endWork();
+            resolve();  // Reject somewhere?
```

Comment:
> `// Reject somewhere?`

You want to `reject()` iff there's an error that the caller could `.catch()` and do something with. What might be an error here? Where would it come from?